cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -594,6 +594,17 @@ def master_services_down():
             failing_services.append(service)
     return failing_services
 
+def add_systemd_file_limit():
+    directory = '/etc/systemd/system/snap.kube-apiserver.daemon.service.d'
+    file_name = 'file-limit.conf'
+    path = os.path.join(directory, file_name)
+    if not os.path.isfile(path):
+        with open(path, 'w') as f:
```

Comment:
I don't believe this will create the directory if it doesn't exist, will it?

Reply:
Yeah, the assumption is that that directory exists because the snap has been installed.  Can you see a scenario where it doesn't?

Reply:
I can't see a scenario, but I just like to guard. Snaps could change, systemd could change, etc. The other addon we are doing ensures the directory exists beforehand as well. I'm flexible here but would lean on the side of safety if pushed.